File: app/src/main/java/com/fyp/biddingapp/Models/SharedPreferenceManager.java
package com.fyp.biddingapp.Models;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;

import com.android.volley.RequestQueue;

public class SharedPreferenceManager {
    private RequestQueue requestQueue;
    @SuppressLint("StaticFieldLeak")
    private static Context ctx;
    @SuppressLint("StaticFieldLeak")
    private static SharedPreferenceManager instance;

    private static final String SHARED_PREF_NAME = "mysharedpref12";
    private static final String KEY_ID = "userId";
    private static final String KEY_NAME = "userName";
    private static final String KEY_EMAIL = "email ";

    private SharedPreferenceManager(Context context) {
        ctx = context;


    }

    public static synchronized SharedPreferenceManager getInstance(Context context) {
        if (instance == null) {
            instance = new SharedPreferenceManager(context);
        }
        return instance;
    }

    public boolean UserLogin (int userId , String UserName , String Email)
    {
        SharedPreferences sharedPreferences = ctx.getSharedPreferences(SHARED_PREF_NAME , Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();

        editor.putInt(KEY_ID , userId);
        editor.putString(KEY_NAME , UserName);
        editor.putString(KEY_EMAIL , Email);
        editor.apply();
        return true ;
    }

    public boolean loggedin ()
    {
        SharedPreferences sharedPreferences = ctx.getSharedPreferences(SHARED_PREF_NAME , Context.MODE_PRIVATE);
        if (sharedPreferences.getString(KEY_NAME , null)!= null)
        {
            return  true ;
        }
        return false ;
    }


    public  boolean logout()
    {
        SharedPreferences sharedPreferences = ctx.getSharedPreferences(SHARED_PREF_NAME , Context.MODE_PRIVATE);
        SharedPreferences.Editor  editor = sharedPreferences.edit();
        editor.clear();
        editor.apply();

        return true;

    }

    public int getUserID ()
    {
        SharedPreferences sharedPreferences = ctx.getSharedPreferences(SHARED_PREF_NAME , Context.MODE_PRIVATE);
        return sharedPreferences.getInt(KEY_ID , 0);
    }

    public String getUserName ()
    {
        SharedPreferences sharedPreferences = ctx.getSharedPreferences(SHARED_PREF_NAME , Context.MODE_PRIVATE);
        return sharedPreferences.getString(KEY_NAME , null);
    }



}

File: app/src/main/java/com/fyp/biddingapp/Models/SharedPreferenceManager.kt
package com.fyp.biddingapp.Models

import com.android.volley.RequestQueue
import android.content.SharedPreferences
import com.fyp.biddingapp.Models.SharedPreferenceManager
import android.annotation.SuppressLint
import android.content.Context
import kotlin.jvm.Synchronized

class SharedPreferenceManager private constructor(@field:SuppressLint("StaticFieldLeak") private var ctx: Context) {
    private val requestQueue: RequestQueue? = null
    fun UserLogin(userId: Int, UserName: String?, Email: String?): Boolean {
        val sharedPreferences: SharedPreferences = Companion.ctx.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putInt(KEY_ID, userId)
        editor.putString(KEY_NAME, UserName)
        editor.putString(KEY_EMAIL, Email)
        editor.apply()
        return true
    }

    fun loggedin(): Boolean {
        val sharedPreferences: SharedPreferences = Companion.ctx.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE)
        return if (sharedPreferences.getString(KEY_NAME, null) != null) {
            true
        } else false
    }

    fun logout(): Boolean {
        val sharedPreferences: SharedPreferences = Companion.ctx.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.clear()
        editor.apply()
        return true
    }

    val userID: Int
        get() {
            val sharedPreferences: SharedPreferences = Companion.ctx.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE)
            return sharedPreferences.getInt(KEY_ID, 0)
        }

    companion object {
        @SuppressLint("StaticFieldLeak")
        private var instance: SharedPreferenceManager? = null
        private const val SHARED_PREF_NAME = "mysharedpref12"
        private const val KEY_ID = "userId"
        private const val KEY_NAME = "userName"
        private const val KEY_EMAIL = "email "
        @Synchronized
        fun getInstance(context: Context): SharedPreferenceManager? {
            if (instance == null) {
                instance = SharedPreferenceManager(context)
            }
            return instance
        }
    }
}